CHANGELOG.md

```diff
@@ -19,6 +19,150 @@ const expandWidget = new Expand({
+```js
+// prompts a deprecation warning if popup isn't created
+// calls view.openPopup() under the hood
+view.popup.open(...);
+
+// new
+view.openPopup(...);
+```
+- It is recommended to use the new `View.closePopup()` method instead of `view.popup.close()`, however, it will still work.
+```js
+// prompts a deprecation warning if popup isn't created
+// calls view.closePopup() under the hood
+view.popup.close(...);
+
+// new
+view.closePopup();
+```
+
+#### Disable the view's popup
+```js
+// autoOpenEnabled deprecated
+view.popup.autoOpenEnabled = true;
+
+// new
+view.popupEnabled = true;
+```
+
+### Breaks and require immediate attention
+
+Using functions other than `open` and `close`, like `watch`. Use `reactiveUtils` instead.
+```js
+// old - will break
+view.popup.watch("selectedFeature", ...)
+
+// new
+reactiveUtils.watch(() => view.popup?.selectedFeature, ...);
+```
+
+Modifying properties that don't exist anymore.
+```js
+// old - will break
+view.popup.actions.push(...)
+
+// or define the property with the actions
+view.popup.actions = [ ... ]
+// new: or wait for the popup
+reactiveUtils.whenOnce(() => view.popup.actions != null);
+view.popup.actions.push(...)
+```
+
+Same as above, `viewModel` doesn't exist.
+
+```js
+// old - will break
+get active() {
+  return view.popup.viewModel.active;
+}
+
+// new
+get active() {
+  return view.popup?.viewModel?.active ?? false;
+}
+```
+
+### Migration Strategy
+
+#### When do you have to migrate, and what to do?
+
+1. If the application creates a `Popup` instance, no changes are necessary. This is also **the fastest migration strategy**, and equivalent to the 4.26 behavior.
+
+```js
+const view = new SceneView({
+  popup: new Popup(...)
+});
+```
+
+2. If the application accesses properties of the view model, use conditionally access.
+
+```js
+view.popup.viewModel.active;
+// becomes
+view.popup?.viewModel?.active;
+```
+
+3. If the application uses `view.popup.open()` or `close()`, search and replace with `view.openPopup()` and `view.closePopup()`.
+
+4. If the application is only setting options on the popup, no changes are necessary.
+
+```js
+const view = new MapView({
+  popup: {
+    dockEnabled: true.
+    dockPosition: ....
+  }
+})
+```
+
+### Configuring the `Popup`
+
+There are multiple ways to configure the popup:
+
+```js
+// popup shows for features and other use such as Search results.
+// view.popup is available immediately
+view.popup = new Popup() 
+        
+//  popup shows for features and other use such as Search results.
+// view.popup is created on demand when the user clicks on the view, or when `openPopup()` is called the first time.
+view.popup = { /*...*/ };
+
+// popup is completely disabled and won't show on features, Search results, and `openPopup()`.
+view.popup = null;
+
+// popup doesn't show for features on click, but will on Search results and `openPopup()`.
+view.popupEnabled = false;
+```
+</details>
+
 ## Breaking Changes
 
 * TBD
```

Comment:
Should there be something in the breaking changes section about this?

Reply:
Yes :)